src/util/containers/SingleRootContainerManager.ts

```diff
@@ -0,0 +1,35 @@
+import type { ResourceIdentifier } from '../../ldp/representation/ResourceIdentifier';
+import { InternalServerError } from '../errors/InternalServerError';
+import { ensureTrailingSlash } from '../PathUtil';
+import type { ContainerManager } from './ContainerManager';
+
+/**
+ * A ContainerManager that assumes there is only 1 root and all other identifiers are made by appending to that root.
+ */
+export class SingleRootContainerManager implements ContainerManager {
+  private readonly baseUrl: string;
+
+  public constructor(baseUrl: string) {
+    this.baseUrl = ensureTrailingSlash(baseUrl);
+  }
+
+  public isValidIdentifier(identifier: ResourceIdentifier): boolean {
+    return identifier.path.startsWith(this.baseUrl);
+  }
+
+  public getParentContainer(identifier: ResourceIdentifier): ResourceIdentifier {
+    // Trailing slash is necessary for URL library
+    const parentPath = new URL('..', ensureTrailingSlash(identifier.path)).href;
+
+    // This probably means there is an issue with the root
```

Comment:
I don't get this comment.

Reply:
Just bad phrasing, but meant that something went wrong if the parent container of the root is being requested.